### PR TITLE
Supply req, res to health check method

### PR DESCRIPTION
**Pull request by martintajur**

### CODE REVIEW

**Comment by martintajur** [comment]
@vadzappa Unit tests and some fixes missing still. Hold on, I will let you know when it's ready for review.

**Comment by martintajur** [comment]
Ready for review @vadzappa

File: lib/status.js [hunk 1]
```diff
@@ -307,16 +309,27 @@
 		};
 
 		checkTimeout = function() {
-			callback(new Error('Health check timed out (> ' + this.healthcheckTimeout + ' sec)'));
+			return callback(new Error('Health check timed out (> ' + this.healthcheckTimeout + ' sec)'));
 		};
+		
+		var parsedUrl = URL.parse(req.url),
```
Comment (vadzappa) [inline lib/status.js]: you can actually do parsing in one step using:
https://nodejs.org/docs/latest-v0.10.x/api/url.html#url_url_parse_urlstr_parsequerystring_slashesdenotehost
just provide true as second argument and QS params will be parsed as well.

File: lib/status.js [hunk 2]
```diff
@@ -307,16 +309,27 @@
...
+			urlParams = null;
+
+		if (parsedUrl.query) {
+			urlParams = querystring.parse(parsedUrl.query);
+		}
+
+		if (urlParams && urlParams.service_name && urlParams.service_name !== this.config.name) {
+			return callback(new Error('Service name mismatch'));
```
Comment (vadzappa) [inline lib/status.js]: not sure how usefull this might be, but what if we'd include both expected and real service names into error message?

**Pull request merged**